### PR TITLE
Reorganize options in man pages

### PR DESCRIPTION
**Pull request by honggyukim**
This PR reorganizes options for each command as follows:
- Separate target command only options, common options, and common analysis options.
- Fix the order of options same across different commands.
- Update/remove/add some incorrect options.

By having different option categories, users may be able to pick useful options for each command.

### CODE REVIEW

**Comment by honggyukim** [comment]
Hi @namhyung, this makes a huge changes, so it may be better to review it using diff tools across each command rather than examining diff changes.

File: doc/uftrace-live.md [hunk 1]
```diff
@@ -50,6 +46,39 @@ OPTIONS
...
+    a `trace_on` trigger.
+
+
+LIVE OPTIONS
+============
+\--list-event
+:   Show available events in the process.
+
+\--report
+:   Show live-report before replay.
```
Comment (honggyukim) [inline doc/uftrace-live.md]: I'm not so sure if they have to be in another category, but just wanted to talk about it what to do.

File: doc/uftrace-record.md [hunk 2]
```diff
@@ -21,12 +21,87 @@ This data can then be inspected later on, using `uftrace replay` or
...
+:   Set kernel max function depth separately.  Implies `--kernel`.
 
+-H *HOST*, \--host=*HOST*
+:   Send trace data to given host via the network, not writing to files.
+    The `uftrace recv` command should be run on the destination host to receive
+    the data.
+
+\--port=*PORT*
+:   When sending data to the network (with `-H`), use the given port instead of
+    the default (8090).
```
Comment (honggyukim) [inline doc/uftrace-record.md]: I'm not sure yet, but I think `--host 127.0.0.1` and `--port 8888` options can be easily used as `--host 127.0.0.1:8888`. We can consider it later.

File: doc/uftrace-live.md (hunk 1 above)
Comment (namhyung) [reply doc/uftrace-live.md]: I'm fine with the current state.

File: doc/uftrace-record.md [hunk 3]
```diff
@@ -375,16 +376,11 @@ The `traceon` and `traceoff` actions (the `_` can be omitted from `trace_on`
...
 The 'recover' trigger is for some corner cases in which the process accesses the
-callstack directly.  During tracing of the v8 javascript engine, for example, it
-kept getting segfaults in the garbage collection stage.  It was because v8
-incorporates the return address into compiled code objects(?).  The `recover`
-trigger restores the original return address at the function entry point and
-resets to the uftrace return hook address again at function exit.  I was managed
-to work around the segfault by setting the `recover` trigger on the related
-function (specifically `ExitFrame::Iterate`).
+callstack directly.  For now it's not necessary to call it as uftrace does the
+job automatically.
```
Comment (namhyung) [inline doc/uftrace-record.md]: This applies to x86 only.  It's hard to do the same on ARM/AArch64 at least currently..

File: doc/uftrace-live.md [hunk 4]
```diff
@@ -39,21 +39,21 @@ COMMON OPTIONS
...
 :   Do not record library function invocations.  Library calls are normally
-    traced by hooking the dynamic linker's resolve function in the PLT.  One can
-    disable it with this option.
+    traced by hooking the dynamic linker's resolve function in the PLT.
+    One can disable it with this option.
 
 \--no-event
 :   Disable event recording which is used by default.  Note that explicit event
-    tracing by \--event option is not affected by this.
+    tracing by `--event` option is not affected by this.
```
Comment (namhyung) [inline doc/uftrace-live.md]: I think it's better to use "\\" for escape rather than enclosing with "\`"

Comment (honggyukim) [reply doc/uftrace-live.md]: I will take a look later for many other places.

File: doc/uftrace-record.md (hunk 2 above)
Comment (namhyung) [reply doc/uftrace-record.md]: Yes, that should be possible too

**Pull request merged**